tools/wpt/browser.py

```diff
@@ -202,8 +238,14 @@ def get_profile_bundle_url(self, version, channel):
 
         return "%s/archive/%s.zip/testing/profiles/" % (repo, tag)
 
-    def install_prefs(self, binary, dest=None):
-        version, channel = self.get_version_and_channel(binary)
+    def install_prefs(self, binary, dest=None, channel=None):
+        version, channel_ = self.get_version_and_channel(binary)
+        if channel is not None and channel != channel_:
+            # Beta doesn't always seem to have the b in the version string, so allow the
+            # manually supplied value to override the one from the binary
```

Comment:
Wow, really? This is good to know. It's yet another reason why we need to include channels in `wptreport.json`. Currently the receiver infers the channel from the version string, which according to this comment is apparently unreliable.